lockbox-ios/Presenter/ItemDetailPresenter.swift

```diff
@@ -0,0 +1,118 @@
+}
+
+class ItemDetailPresenter {
+    weak var view:ItemDetailViewProtocol?
+    private var dataStore:DataStore
+    private var itemDetailStore:ItemDetailStore
+    private var routeActionHandler:RouteActionHandler
+    private var itemDetailActionHandler:ItemDetailActionHandler
+    private var disposeBag = DisposeBag()
+
+    lazy private(set) var onPasswordToggle:AnyObserver<Void> = {
+        return Binder(self) { target, _ in
+            let updatedPasswordReveal = target.view?.passwordRevealed ?? false
+            self.itemDetailActionHandler.invoke(.togglePassword(displayed: updatedPasswordReveal))
+        }.asObserver()
+    }()
+
+    lazy private(set) var onCancel:AnyObserver<Void> = {
+        return Binder(self) { target, _ in
+            self.routeActionHandler.invoke(MainRouteAction.list)
+        }.asObserver()
+    }()
+
+    init(view:ItemDetailViewProtocol,
+         dataStore:DataStore = DataStore.shared,
+         itemDetailStore:ItemDetailStore = ItemDetailStore.shared,
+         routeActionHandler:RouteActionHandler = RouteActionHandler.shared,
+         itemDetailActionHandler:ItemDetailActionHandler = ItemDetailActionHandler.shared) {
+        self.view = view
+        self.dataStore = dataStore
+        self.itemDetailStore = itemDetailStore
+        self.routeActionHandler = routeActionHandler
+        self.itemDetailActionHandler = itemDetailActionHandler
+
+        self.itemDetailActionHandler.invoke(.togglePassword(displayed: false))
+    }
+
+    func onViewReady() {
+        let itemObservable = self.dataStore.onItem(self.view?.itemId ?? "")
+
+        let viewConfigDriver = Driver.combineLatest(itemObservable.asDriver(onErrorJustReturn: Item.Builder().build()), self.itemDetailStore.itemDetailDisplay)
+                .map { e -> [ItemDetailSectionModel] in
+                    if case let .togglePassword(passwordDisplayed) = e.1 {
+                        return self.configurationForItem(e.0, passwordDisplayed: passwordDisplayed)
+                    }
+
+                    return self.configurationForItem(e.0, passwordDisplayed: false)
+                }
+
+        let titleDriver = itemObservable
+                .map { item -> String in
+                    return item.title ?? item.origins.first ?? Constant.string.unnamedEntry
+                }.asDriver(onErrorJustReturn: Constant.string.unnamedEntry)
+
+        self.view?.bind(itemDetail: viewConfigDriver)
+        self.view?.bind(titleText: titleDriver)
+    }
+}
+
+// helpers
+extension ItemDetailPresenter {
+    private func configurationForItem(_ item: Item, passwordDisplayed:Bool) -> [ItemDetailSectionModel] {
+        var passwordText:String
+        let itemPassword:String = item.entry.password ?? ""
+
+        if passwordDisplayed {
+            passwordText = itemPassword
+        } else {
+            passwordText = itemPassword.replacingOccurrences(of: "[^\\s]", with: "•", options: .regularExpression)
```

Comment:
Right now, the value components are all `UILabel`s, not `UITextField`s, so there's no  -- editing or adding entries isn't in the upcoming roadmap at all, we're just going to do readonly ones to start. However, there's an argument to be made for having these fields all be text fields with `userInteractionEnabled = false` in anticipation of that work (& to avoid having to do the text munging on the fly).

Reply:
per conversation with @sandysage , I'm going to leave these as `UILabels` for now, editing isn't in our roadmap at all right so I don't see a justification in converting to `UITextField`.

Reply:
Hmm. I liked your plan to use uneditable `UITextField`, even if you never made them editable.

Also q.v. No Plan Survives First Contact with the Enemy

Reply:
`[^\\s]` is a character that is not whitespace. 
`\\S` is too.